system/HTTP/SiteURI.php

```diff
@@ -363,4 +363,67 @@ protected function applyParts(array $parts): void
             $this->password = $parts['pass'];
         }
     }
+
+    /**
+     * For baser_url() helper.
+     *
+     * @param array|string $relativePath URI string or array of URI segments
+     * @param string|null  $scheme       URI scheme. E.g., http, ftp
+     */
+    public function baseUrl($relativePath = '', ?string $scheme = null): string
+    {
+        $relativePath = $this->stringifyRelativePath($relativePath);
+
+        $config            = clone config(App::class);
+        $config->indexPage = '';
+
+        $host = $this->getHost();
+
+        $uri = new self($config, $relativePath, $host, $scheme);
+
+        // Support protocol-relative links
+        if ($scheme === '') {
```

Comment:
Should this handle `null` as well, like above?

Reply:
No. `null` is the default value for `$scheme`, and `base_url()` must return the baseURL (e.g., `http://example.com/`).
But `''` is for protocol-relative links (`//example.com/img/test.jpg`). It must omit the protocol part.